tests/test_ankaios.py

```diff
@@ -72,229 +70,54 @@ def test_logger():
-
-        # Start thread (similar to _connect)
-        ankaios._read_thread = threading.Thread(
-            target=ankaios._read_from_control_interface
-        )
-        ankaios._connected = True
-        ankaios._read_thread.start()
-        time.sleep(0.01)
-
-        # Stop thread (similar to disconnect)
-        ankaios._connected = False
-        ankaios._disconnect_event.set()
-        ankaios._read_thread.join()
-
-        mock_file.assert_called_once_with(
-            "/run/ankaios/control_interface/input", "rb")
-        assert "1234" in list(ankaios._responses)
-        assert ankaios._responses["1234"].is_set()
-
-    # Test request set first
-    with patch("builtins.open", mock_open()) as mock_file, \
-            patch("os.set_blocking") as _, \
-            patch("select.select") as mock_select:
-        mock_select.return_value = ([True], [], [])
-        mock_file_handle = mock_file.return_value.__enter__.return_value
-        mock_file_handle.read.side_effect = \
-            [bytes([b]) for b in input_file_content]
-
-        ankaios = generate_test_ankaios()
-        ankaios._responses["1234"] = ResponseEvent()
-
-        # Start thread (similar to _connect)
-        ankaios._read_thread = threading.Thread(
-            target=ankaios._read_from_control_interface
-        )
-        ankaios._connected = True
-        ankaios._read_thread.start()
-        time.sleep(0.01)
-
-        # Stop thread (similar to disconnect)
-        ankaios._connected = False
-        ankaios._disconnect_event.set()
-        ankaios._read_thread.join()
-
-        mock_file.assert_called_once_with(
-            "/run/ankaios/control_interface/input", "rb")
-        assert "1234" in list(ankaios._responses)
-        assert ankaios._responses["1234"].is_set()
+    with patch("ankaios_sdk.ControlInterface.connect") as mock_ci_connect, \
+         patch("ankaios_sdk.ControlInterface.disconnect") \
+            as mock_ci_disconnect:
+        with Ankaios() as ankaios:
+            assert isinstance(ankaios, Ankaios)
+            mock_ci_connect.assert_called_once()
+            assert ankaios.logger.level == AnkaiosLogLevel.INFO.value
+        mock_ci_disconnect.assert_called_once()
 
 
-def test_write_to_pipe():
+def test_state():
     """
-    Test the _write_to_pipe method of the Ankaios class.
+    Test the state property of the Ankaios class and the state callback.
     """
     ankaios = generate_test_ankaios()
-    with pytest.raises(AnkaiosConnectionException,
-                       match="Could not write to pipe, not connected."):
-        ankaios._write_to_pipe(None)
-
-    ankaios._connected = True
-    output_file = MagicMock()
-    ankaios._output_file = output_file
-
-    ankaios._write_to_pipe(MESSAGE_UPDATE_SUCCESS)
-
-    output_file.write.assert_called()
-    output_file.flush.assert_called_once()
+    assert ankaios.state == ankaios._control_interface._state
+    ankaios._state_changed(ControlInterfaceState.DISCONNECTED)
```

Comment:
Yes the property check is fine, but for now the second method call is unnecessary because it does not fail when changing something in there. You can assert the log to have at least one assertion for the _state_changed() call.

Reply:
Done.